common/client.go

```diff
@@ -365,14 +369,43 @@ func (rc *RestClient) execMethod(method string, dest string, data interface{}, r
 	if err != nil {
 		return err
 	}
-
-	if result == nil {
-		return nil
+	var unmarshalBodyErr error
+	if result != nil {
+		unmarshalBodyErr = json.Unmarshal(body, &result)
 	}
-
-	err = json.Unmarshal(body, &result)
-	if err != nil {
-		return errors.New(fmt.Sprintf("Error %s (%T) when parsing %s", err.Error(), err, body))
+	if resp.StatusCode >= 400 {
```

Comment:
The most immediately self-documenting way is to have a function called isHttpError(), so that you can simply write:

    if isHttpError(resp.StatusCode) {
        ...

That documents the intent here and wherever the function is written, its name documents the code there.